### PR TITLE
Add functionality to check image directly from host machine 

### LINKED ISSUE #112: Pulling an image INFO message is misleading
**Describe the bug**
`INFO[2019-07-19 23:39:20] Pulling an image: 'golang'` This message is misleading when the system already has the Docker image pulled.

**To Reproduce**
Steps to reproduce the behaviour:
1. Create a `.dunner.yaml`
2. Add a task
3. Run the task ( This will pull the image )
4. Run the task again

**Expected behaviour**
Dunner should not pull the image again nor show an INFO message saying pulling image.

**Screenshots**
Screenshot from 2019-07-19 23-42-52

### PR DESCRIPTION
**Pull request by ayushjn20**
Resolves #112 and #114

### CODE REVIEW

File: pkg/docker/docker.go [hunk 1]
```diff
@@ -222,3 +229,31 @@ func ExtractResult(reader io.Reader, command []string) *Result {
 	}
 	return &result
 }
+
+func checkImageExist(ctx context.Context, cli *client.Client, image string, notag bool) bool {
+	log.Debugf("docker: checking existence of the image '%s'", image)
```
Comment (agentmilindu) [inline pkg/docker/docker.go]: Should be `"Docker: Checking exisitence"`

File: pkg/docker/docker.go [hunk 2]
```diff
@@ -222,3 +229,31 @@ func ExtractResult(reader io.Reader, command []string) *Result {
 	}
 	return &result
 }
+
+func checkImageExist(ctx context.Context, cli *client.Client, image string, notag bool) bool {
```
Comment (apoorvam) [inline pkg/docker/docker.go]: There are many conditionals in this and unit tests will help understand better.

File: pkg/docker/docker.go (hunk 1 above)
Comment (ayushjn20) [reply pkg/docker/docker.go]: https://blog.golang.org/error-handling-and-go
Check this^ blog out. Though the convention is defined for error statements, I have applied similar to debug statements as well to maintain the consistency.

Comment (agentmilindu) [reply pkg/docker/docker.go]: Have they mentioned about the case?

File: pkg/docker/docker.go (hunk 2 above)
Comment (agentmilindu) [reply pkg/docker/docker.go]: @ayushjn20 Did you add unit tests?

**Comment by agentmilindu** [comment]
@ayushjn20 Resolve the conflicts

File: Makefile [hunk 3]
```diff
@@ -43,6 +45,9 @@ lint:
 
 precommit: build test fmt lint vet
 
+test-setup:
+	@docker build --no-cache --tag 'test-image' $(TEST_IMAGE_DIR)
```
Comment (agentmilindu) [inline Makefile]: How about we change this into `dunner/test-image`, to avoid possible conflicts?

Comment (ayushjn20) [reply Makefile]: Done

File: pkg/docker/docker.go (hunk 2 above)
Comment (ayushjn20) [reply pkg/docker/docker.go]: Done

**Comment by apoorvam** [comment]
Codacy pointed an enhancement here: https://app.codacy.com/app/LeopardsLab/Dunner/pullRequest?prid=3891532 @ayushjn20

**Pull request merged**